### PR TITLE
Hdr.mergemertens.fixalgo

### PR DESCRIPTION
**Pull request by cedrou**
Fix for #5807: 
> In the MergeMertensImpl::process method, the implementation of the formula to compute the well-exposedness map is false and is missing a call to cv::exp to get a gaussian curve.

### CODE REVIEW

File: modules/photo/include/opencv2/photo.hpp [hunk 1]
```diff
@@ -694,7 +694,7 @@ class CV_EXPORTS_W MergeMertens : public MergeExposures
 @param exposure_weight well-exposedness measure weight
  */
 CV_EXPORTS_W Ptr<MergeMertens>
-createMergeMertens(float contrast_weight = 1.0f, float saturation_weight = 1.0f, float exposure_weight = 0.0f);
+createMergeMertens(float contrast_weight = 1.0f, float saturation_weight = 1.0f, float exposure_weight = 1.0f);
```
Comment (alalek) [inline modules/photo/include/opencv2/photo.hpp]: Could you please revert this change, because it breaks source code compatibility (and breaks tests).

If current default parameters make no sense, you can mark this to be fixed in the next major release (current we don't maintain "next" branch):
```
#ifndef OPENCV_ABI_COMPATIBILITY
CV_EXPORTS_W Ptr<MergeMertens>
createMergeMertens(float contrast_weight = 1.0f, float saturation_weight = 1.0f, float exposure_weight = 1.0f);
#else
CV_EXPORTS_W Ptr<MergeMertens>
createMergeMertens(float contrast_weight = 1.0f, float saturation_weight = 1.0f, float exposure_weight = 0.0f);
#endif
```

**Comment by cedrou** [comment]
OK, I've just reverted the commit. 
The current default parameters are not wrong, the point is that it is not the default values recommended by the author of the algorithm.

**Comment by alalek** [comment]
BTW, Current failures is not related to this PR

**Comment by vpisarev** [comment]
thanks! :+1:
